Adapter.C

```diff
@@ -651,8 +651,8 @@ void preciceAdapter::Adapter::adjustSolverTimeStepAndReadData()
        If the solver tries to use a bigger timestep, then it needs to use
        the same timestep as the one determined by preCICE.
     */
-
-    if (timestepSolverDetermined < timestepPrecice_)
+    double tolerance = 1e-12;
```

Comment:
Why e-12 if preCICE sets e-14?
We should probably add a comment.

Ideally, preCICE should provide this.

Reply:
> Ideally, preCICE should provide this.

If you think this is valuable, please comment here https://github.com/precice/precice/issues/1902

Overall, we just use it to inform the user and I don't see a specific need or reason to ask preCICE for this.

> Why e-12 if preCICE sets e-14?

Putting it the other way around: for which differences do you want to emit a warning? I just picked what I thought could be of practical relevance, but we can definitely change it.

Reply:
Does the issue originate primarily from error accumulation in preCICE, or in OpenFOAM? Is it controlled by any accuracy set (at compile-time or runtime) by preCICE?

If it depends on the accuracy set in preCICE, then it just sounds reasonable to me to use that directly here, as one can then make the connection.

But since this only affects warnings and the behavior does not change, then it does not really matter.

No need to change anything in this context.

Reply:
It is a compile time constant in preCICE, but there is no guarantee that our time-step computation is accurate up to these many digits, but ok then let me pick 14.